c_glib/parquet-glib/arrow-file-writer.cpp

```diff
@@ -496,6 +495,45 @@ gparquet_arrow_file_writer_new_path(GArrowSchema *schema,
   }
 }
 
+/**
+ * gparquet_arrow_file_writer_get_schema:
+ * @writer: A #GParquetArrowFileWriter.
+ *
+ * Returns: (transfer full): The schema to be written to.
+ *
+ * Since: 18.0.0
+ */
+GArrowSchema *
+gparquet_arrow_file_writer_get_schema(GParquetArrowFileWriter *writer)
+{
+  auto parquet_arrow_file_writer = gparquet_arrow_file_writer_get_raw(writer);
+  auto arrow_schema = parquet_arrow_file_writer->schema();
+  return garrow_schema_new_raw(&arrow_schema);
+}
+
+/**
+ * gparquet_arrow_file_writer_write_record_batch:
+ * @writer: A #GParquetArrowFileWriter.
+ * @record_batch: A record batch to be written.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: %TRUE on success, %FALSE if there was an error.
+ *
+ * Since: 18.0.0
+ */
+gboolean
+gparquet_arrow_file_writer_write_record_batch(GParquetArrowFileWriter *writer,
+                                              GArrowRecordBatch *record_batch,
+                                              GError **error)
+{
+  auto parquet_arrow_file_writer = gparquet_arrow_file_writer_get_raw(writer);
+  auto arrow_record_batch = garrow_record_batch_get_raw(record_batch).get();
+  auto status = parquet_arrow_file_writer->WriteRecordBatch(*arrow_record_batch);
```

Comment:
Thanks for using this!

The writeRecordBatch is added by @wgtmac . Notice that this will not flush the RowGroup ( WriteTable will switch rg )

Reply:
Oh, I didn't notice it.
Should we also add the bindings of `parquet::arrow::FileWriter::NewRowGroup()` (and `NewBufferedRowGroup()`?) for it?

Reply:
It doesn't matter, currently we can control the row-group flush by size, just a notice that they're different. NewRowGroup()/NewBufferedRowGroup is also ok in this scenerio

Reply:
Ah, OK.
I should have read the `WriteRecordBatch()` implementation more before I asked.
It calls `NewBufferedRowGroup()` automatically:
https://github.com/apache/arrow/blob/d88dd19ea83aee37a4da69bdbc688f5e342fb455/cpp/src/parquet/arrow/writer.cc#L425
https://github.com/apache/arrow/blob/d88dd19ea83aee37a4da69bdbc688f5e342fb455/cpp/src/parquet/arrow/writer.cc#L468

Anyway, I'll add the bindings of them for advanced use cases: GH-44006, GH-44007

Reply:
Yeah the difference is that https://github.com/apache/arrow/blob/d88dd19ea83aee37a4da69bdbc688f5e342fb455/cpp/src/parquet/arrow/writer.cc#L382

The writeTable will forcing NewRowGroup, but writeBatch will not...